core/src/epoch_slots.rs

```diff
@@ -1,24 +1,52 @@
 use crate::cluster_info::MAX_CRDS_OBJECT_SIZE;
+use crate::crds_value::MAX_SLOT;
+use crate::crds_value::MAX_WALLCLOCK;
 use bincode::serialized_size;
 use bv::BitVec;
 use flate2::{Compress, Compression, Decompress, FlushCompress, FlushDecompress};
 use solana_sdk::clock::Slot;
 use solana_sdk::pubkey::Pubkey;
+use solana_sdk::sanitize::{Sanitize, SanitizeError};
 
+const MAX_SLOTS_PER_ENTRY: usize = 2048 * 8;
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 pub struct Uncompressed {
     pub first_slot: Slot,
     pub num: usize,
     pub slots: BitVec<u8>,
 }
 
+impl Sanitize for Uncompressed {
+    fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
+        if self.first_slot >= MAX_SLOT {
+            return Err(SanitizeError::ValueOutOfRange);
+        }
+        if self.num >= MAX_SLOTS_PER_ENTRY {
+            return Err(SanitizeError::ValueOutOfRange);
+        }
+        Ok(())
+    }
+}
+
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
 pub struct Flate2 {
     pub first_slot: Slot,
     pub num: usize,
     pub compressed: Vec<u8>,
 }
 
+impl Sanitize for Flate2 {
+    fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
+        if self.first_slot >= MAX_SLOT {
+            return Err(SanitizeError::ValueOutOfRange);
+        }
+        if self.num >= MAX_SLOTS_PER_ENTRY {
```

Comment:
similar here, should we have a check `self.num < 3 * self.slots.len()`

Reply:
is that the max compression ratio?  id rather have a simple hard check for the wire stuff

Reply:
I wouldn't say it's the max. 3x is a decent ratio for somewhat compressible data. You could go higher like 4-5x. Of course you can create pathological cases where your compression has a 1000x ratio. I was trying to avoid those cases. Basically a mini zip bomb through a gossip packet.

Reply:
we need a inflate that guards against the max

Reply:
nm, the way this is written, the sanitize check will guard num, and then the vector allocation cannot acceed num.

Reply:
the guard num is huge though

Reply:
The guard seems to allow for 16k * ~100-ish entries ~= 1.5MB/packet. Attacker could send a bunch of packets. Maybe it's ok.